### PR TITLE
modify: don't require . or / for starting char, don't delete on failure

### PR DESCRIPTION
**Pull request by dgonyeo**
When the modify flag was used, it was requiring that the initial
character for the path provided start with a `.` or a `/`. Since the
modify flag can only be used on local images, this was unnecessary.

Also when using the modify flag and the write failed (e.g. due to
permissions issues), the original image was deleted. This commit changes
it so that the write happens to a different file from the original, and
on success replaces the original with the new image.

### CODE REVIEW

File: acbuild/acbuild.go [hunk 1]
```diff
@@ -131,6 +131,16 @@ func runWrapper(cf func(cmd *cobra.Command, args []string) (exit int)) func(cmd
 			return
 		}
 
+		if len(aciToModify) == 0 {
```
Comment (jonboulle) [inline acbuild/acbuild.go]: how could this ever happen? probably need to panic (see 118)

File: acbuild/acbuild.go [hunk 2]
```diff
@@ -131,6 +131,16 @@ func runWrapper(cf func(cmd *cobra.Command, args []string) (exit int)) func(cmd
...
+			stderr("Path to ACI to modify can't be empty.")
+			cmdExitCode = 1
+			return
+		}
+
+		if aciToModify[0] != '.' && aciToModify[0] != '/' {
+			aciToModify = "./" + aciToModify
```
Comment (jonboulle) [inline acbuild/acbuild.go]: wait why is this necessary?

Comment (jonboulle) [inline acbuild/acbuild.go]: oh for begin? hmm. it's a little awkward

Comment (jonboulle) [inline acbuild/acbuild.go]: How about passing absoluteAciToMoidyf to Begin() instead

File: acbuild/acbuild.go [hunk 3]
```diff
@@ -178,21 +188,36 @@ func runWrapper(cf func(cmd *cobra.Command, args []string) (exit int)) func(cmd
...
+					cmdExitCode = getErrorCode(err)
+				}
+			}
+		}()
+
 		cmdExitCode = cf(cmd, args)
 
-		err = a.Write(aciToModify, true, false, nil)
+		dir, file := path.Split(aciToModify)
+		tmpACIFile := path.Join(dir, "."+file+".tmp")
```
Comment (jonboulle) [inline acbuild/acbuild.go]: any reason to roll your own tempfile here rather than using ioutil.Tempfile() or similar?

Comment (jonboulle) [inline acbuild/acbuild.go]: I guess it's ok, nvm.

File: acbuild/acbuild.go (hunk 2 above)
Comment (dgonyeo) [inline acbuild/acbuild.go]: Yeah, for begin:

```
derek@energia ~/aci> acbuild --modify alpine-latest-linux-amd64.aci set-name dgretg
unknown host when fetching image, check your connection and local file paths must start with '/' or '.'
```

Passing `absoluteAciToModify` works too, I'll do that.

File: acbuild/acbuild.go (hunk 1 above)
Comment (dgonyeo) [inline acbuild/acbuild.go]: Turns out the first thing I do in this function is `if aciToModify == ""`, so this check is redundant. I'll remove it.

**Comment by dgonyeo** [comment]
Anything else you'd like me to tweak before I merge this?

**Comment by jonboulle** [comment]
lgtm
